### PR TITLE
[RV64_DYNAREC] Fixed some bugs for VMP

### PR DESCRIPTION
**Pull request by xctan**

### CODE REVIEW

**Comment by ksco** [comment]
CI is not happy.

**Comment by xctan** [comment]
> CI is not happy.

Fixed.

File: src/dynarec/rv64/dynarec_rv64_helper.h [hunk 1]
```diff
@@ -168,9 +168,9 @@
         ed = i;                                                                               \
         wb1 = 1;                                                                              \
     }
-// GETEW will use i for ed, and can use r3 for wback.
+// GETEW will use i for ed, and *may* use x3 for wback.
 #define GETEW(i, D) GETEWW(x3, i, D)
-// GETSEW will use i for ed, and can use r3 for wback. This is the Signed version
+// GETSEW will use i for ed, and *may* use x3 for wback. This is the Signed version
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_helper.h:173]: Add another  parameter w for wback, so it’s more obvious which register is used.

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_helper.h:173]: Never mind, there are more macros like this across all 3 backends.

**Review by ksco — APPROVED** [review]

File: src/dynarec/rv64/dynarec_rv64_66.c [hunk 2]
```diff
@@ -74,7 +74,7 @@ uintptr_t dynarec64_66(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
             nextop = F8;
             GETGW(x1);
             GETEW(x2, 0);
-            emit_add16(dyn, ninst, x1, x2, x3, x4, x6);
+            emit_add16(dyn, ninst, x1, x2, x5, x4, x6);
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_66.c:77]: GWBACK is fine, but there is no harm in changing either...

**Pull request merged**